src/redux/selectors/index.js

```diff
@@ -1,6 +1,34 @@
-export const getSearchDetails = state => state.imagesGalleryPage
+import { createSelector } from 'reselect'
+
+import { ITEMS_PER_PAGES } from 'common/constants'
+
+// General
+export const getIsLoading = state => state.imgDailyPage.isLoading
+
+//  Image gallery
+export const getImages = state => state.imagesGalleryPage.items
+
+export const getImagesSelector = createSelector(
+  getImages,
+  items => items.flatMap(({ links }) => links),
+)
 
+// Daily image
+export const getDailyImage = state => state.imgDailyPage.image
+
+// Search
+export const getSearchDetails = state => state.imagesGalleryPage
 export const getSearchText = state => state.imagesGalleryPage.searchText
+
+// Pagination
 export const getCurrentPage = state => state.imagesGalleryPage.pageIndex
+export const getTotalItemsOnPage = state => state.imagesGalleryPage.totalItems
+
+export const getTotalItemsOnPageSelector = createSelector(
+  getTotalItemsOnPage,
+  items => items.total_hits / ITEMS_PER_PAGES,
+)
+
+// Years slider
 export const getStartYear = state => state.imagesGalleryPage.yearStart
 export const getEndYear = state => state.imagesGalleryPage.yearEnd
```

Comment:
these 2 as well, you should reuse **imagesGalleryPage** selector in 2 places, instead of destructuring it 2 times

Reply:
please apply the above for all selectors